src/main/scala/com/lucidworks/spark/util/SolrRelationUtil.scala

```diff
@@ -621,7 +669,34 @@ object SolrRelationUtil extends LazyLogging {
               }
             case map: util.Map[_,_] =>
               val obj = map.get(field.name).asInstanceOf[Object]
-              val newValue = processFieldValue(obj, fieldType, multiValued = false)
+              val newValue =  obj match {
```

Comment:
[0] Is this the same ~20 lines that were added above?  If so, would it be easy to pull this into helper function so we only have to maintain this block in one place?